iso8601/parse_test.go

```diff
@@ -0,0 +1,52 @@
+package iso8601
+
+import (
+	"testing"
+	"time"
+)
+
+func TestParse(t *testing.T) {
+	for _, input := range []string{
+		"2006-01-02T15:04:05Z",
+		"2006-01-02T15:04:05.123Z",
+		"2006-01-02T15:04:05.123456Z",
+		"2006-01-02T15:04:05.123456789Z",
```

Comment:
Could we add a couple more tests (maybe generate them?), especially to verify the behavior of the leap year and month adjustment logic with a variety of inputs. A lot could go wrong if the behavior of `iso8601.Parse` doesn't produce the same result as `time.Parse`.